graph/db/interfaces.go

```diff
@@ -23,3 +24,16 @@ type NodeRTx interface {
 	// the same transaction.
 	FetchNode(node route.Vertex) (NodeRTx, error)
 }
+
+// CachedGraph is an abstract read only interface that provides information
+// about nodes and their edges. The interface is about providing fast read-only
+// access to the graph and so if a cache is available, it should be used.
+type CachedGraph interface {
```

Comment:
this is probably a bad name... cause it can _either_ be a cached graph or just direct DB graph. 

so perhaps `NodeTraversal` makes more sense? or something to that tune

Reply:
updated